processing/src/main/java/org/apache/druid/query/aggregation/LongSumAggregatorFactory.java

```diff
@@ -87,8 +89,12 @@ protected VectorAggregator factorizeVector(
   }
 
   @Override
-  public boolean canVectorize()
+  public boolean canVectorize(ColumnCapabilitiesProvider columnCapabilitiesProvider)
   {
+    if (fieldName != null) {
+      final ColumnCapabilities originalCapabilities = columnCapabilitiesProvider.getColumnCapabilities(fieldName);
+      return expression == null && (originalCapabilities == null || originalCapabilities.getType().isNumeric());
+    }
```

Comment:
Which tests validate these conditions? Similar comments for the other AggregatorFactories. I see DoubleMeanAggregator is being tested

Reply:
SchemaEvolutionTest and DoubleMeanAggregationTest with Vectorize.TRUE

Reply:
👍 

Just read `SchemaEvolutionTest` more closely, looks like it covers LongSum, DoubleSum and Count. I didn;t see a test that covers FloatSumAggregatorFactory

Reply:
Done. Added FloatSumAggregatorFactory